src/reducers/index.ts

```diff
@@ -105,19 +105,22 @@ function parseVega(
   };
 }
 
-function parseVegaLite(
-  state: State,
-  action: SetVegaLiteExample | UpdateVegaLiteSpec | SetGistVegaLiteSpec,
-  extend: Partial<State> = {}
-) {
+function parseVegaLite(state: State, action: any, extend: Partial<State> = {}) {
   const currLogger = new LocalLogger();
 
   try {
+    if (!action.spec) {
+      action.spec = state.editorString;
```

Comment:
This also causes a side effect that we don't want.

Reply:
What is the side effect?

Reply:
The code modifies an input. A function should generally avoid doing that.